### PR TITLE
[v1.12] backport of 22179

### PR DESCRIPTION
**Pull request by aanm**
* #22179 -- Add automatic creation of Cilium base images (@aanm)
 * #22268 -- .github/workflows: split the image tag update in two steps (@aanm)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 22179 22268; do contrib/backporting/set-labels.py $pr done 1.12; done
```

### CODE REVIEW

**Comment by aanm** [comment]
/test-backport-1.12

**Review by michi-covalent — APPROVED** [review]
`:hype:`

File: CODEOWNERS [hunk 1]
```diff
@@ -204,6 +204,8 @@
 * @cilium/tophat
```
Comment (joestringer) [inline CODEOWNERS:204]: Not related to this PR, but it looks like the CODEOWNERS file is invalid on this branch. Don't we usually replace the whole codeowners with a more minimal set of owners on older branches...?

**Review by joestringer — APPROVED** [review]
:tada: 

Ack, didn't look too closely.

**Review by christarazi — APPROVED** [review]

**Pull request merged**